Example-Swift/ViewController.swift

```diff
@@ -9,102 +9,69 @@
+    
+    func photosViewController(_ photosViewController: NYTPhotosViewController, handleActionButtonTappedFor photo: NYTPhoto) -> Bool {
+        if UIDevice.current.userInterfaceIdiom == .pad {
             guard let photoImage = photo.image else { return false }
             
             let shareActivityViewController = UIActivityViewController(activityItems: [photoImage], applicationActivities: nil)
-            
-            shareActivityViewController.completionWithItemsHandler = {(activityType: String?, completed: Bool, items: [AnyObject]?, error: NSError?) in
+            shareActivityViewController.completionWithItemsHandler = {(activityType: UIActivityType?, completed: Bool, items: [Any]?, error: Error?) in
                 if completed {
-                    photosViewController.delegate?.photosViewController!(photosViewController, actionCompletedWithActivityType: activityType!)
+                    photosViewController.delegate?.photosViewController!(photosViewController, actionCompletedWithActivityType: activityType?.rawValue)
                 }
             }
 
             shareActivityViewController.popoverPresentationController?.barButtonItem = photosViewController.rightBarButtonItem
-            photosViewController.presentViewController(shareActivityViewController, animated: true, completion: nil)
+            photosViewController.present(shareActivityViewController, animated: true, completion: nil)
 
             return true
         }
         
         return false
     }
     
-    func photosViewController(photosViewController: NYTPhotosViewController, referenceViewForPhoto photo: NYTPhoto) -> UIView? {
-        if photo as? ExamplePhoto == photos[NoReferenceViewPhotoIndex] {
-            return nil
-        }
-        return imageButton
-    }
-    
-    func photosViewController(photosViewController: NYTPhotosViewController, loadingViewForPhoto photo: NYTPhoto) -> UIView? {
-        if photo as! ExamplePhoto == photos[CustomEverythingPhotoIndex] {
-            let label = UILabel()
-            label.text = "Custom Loading..."
-            label.textColor = UIColor.greenColor()
-            return label
-        }
-        return nil
-    }
-    
-    func photosViewController(photosViewController: NYTPhotosViewController, captionViewForPhoto photo: NYTPhoto) -> UIView? {
-        if photo as! ExamplePhoto == photos[CustomEverythingPhotoIndex] {
-            let label = UILabel()
-            label.text = "Custom Caption View"
-            label.textColor = UIColor.whiteColor()
-            label.backgroundColor = UIColor.redColor()
-            return label
+    func photosViewController(_ photosViewController: NYTPhotosViewController, referenceViewFor photo: NYTPhoto) -> UIView? {
+        guard let box = photo as? NYTPhotoBox else { return nil }
+
+        if box.value.name == ReferencePhotoName {
+            return imageButton
         }
+
         return nil
```

Comment:
Done in https://github.com/NYTimes/NYTPhotoViewer/pull/223/commits/141e05bb820ccb27daf361502c2ab4a50751dfda